d3a_interface/data_classes.py

```diff
@@ -150,6 +151,7 @@ def from_dict(offer: Dict) -> "Offer":
             requirements=offer.get("requirements"))
 
     def __eq__(self, other) -> bool:
+        # TODO: maybe add the self.time_slot here, but may make the comparing slower and not needed
```

Comment:
Datetime comparisons are integer comparisons IIRC (essentially only the Unix timestamp is compared IIRC) therefore I would add it too